### ISSUE TITLE
Gossip block validation allows unviable blocks to propagate

### ISSUE THREAD

**Issue by arnetheduck** [ethereum/eth2.0-specs#1959]
https://github.com/ethereum/eth2.0-specs/blob/v0.12.1/specs/phase0/p2p-interface.md#global-topics

We do not ignore blocks that create a new fork from an old (pre-finalization) parent - these block can be produced for example when a node is syncing and is behind the network, but (correctly) creates a block according to its own view of the `head` - by advancing slot of the "old" parent to the current wall clock time.

**Comment by djrtwo** [comment]
@terencechain suggested the following in another issue to be added to the p2p gossip criteria

```python
assert get_ancestor(store, block.parent_root, finalized_slot) == store.finalized_checkpoint.root
```

Makes sense to me. I'll queue this up for v0.12.2

**Comment by protolambda** [comment]
Addressed in #1985
